Stripe/STPCustomer.m

```diff
@@ -85,43 +83,60 @@ + (instancetype)decodedObjectFromAPIResponse:(NSDictionary *)response {
         shipping.country = shippingDict[@"address"][@"country"];
         customer.shippingAddress = shipping;
     }
-    NSMutableArray *sources = [NSMutableArray array];
-    if ([dict[@"sources"] isKindOfClass:[NSDictionary class]] && [dict[@"sources"][@"data"] isKindOfClass:[NSArray class]]) {
-        for (id contents in dict[@"sources"][@"data"]) {
-            if ([contents isKindOfClass:[NSDictionary class]]) {
-                if ([contents[@"object"] isEqualToString:@"card"]) {
-                    STPCard *card = [STPCard decodedObjectFromAPIResponse:contents];
-                    // ignore apple pay cards from the response
-                    if (card && !card.isApplePayCard) {
-                        [sources addObject:card];
-                        if (defaultSourceId && [card.stripeID isEqualToString:defaultSourceId]) {
-                            customer.defaultSource = card;
-                        }
+    [customer updateSourcesWithResponse:dict filteringApplePay:YES];
+    customer.allResponseFields = dict;
+    return customer;
+}
+
+- (void)updateSourcesWithResponse:(NSDictionary *)response
+                filteringApplePay:(BOOL)filterApplePay {
+    NSArray *data;
+    if ([response[@"sources"] isKindOfClass:[NSDictionary class]]) {
+        data = response[@"sources"][@"data"];
+    }
+    if (![data isKindOfClass:[NSArray class]]) {
+        return;
+    }
+    NSString *defaultSourceId;
+    if ([response[@"default_source"] isKindOfClass:[NSString class]]) {
+        defaultSourceId = response[@"default_source"];
+    }
+    NSMutableArray *sources = [NSMutableArray new];
+    for (id contents in data) {
+        if ([contents isKindOfClass:[NSDictionary class]]) {
+            if ([contents[@"object"] isEqualToString:@"card"]) {
```

Comment:
I recognize this is just the old code, but this line makes an assumption about the type of `contents[@"object]"` by using an `NSString`-only method (`isEqualToString:`), where the parsing code so far has been diligently checking types.

Maybe -> `isEqual:@"card"`

Reply:
side note: reading this method makes me want `as?`, `flatMap`, `filter` & `find` :)